src/app/public/testing/dropdown/dropdown-fixture.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  DebugElement
+} from '@angular/core';
+
+import {
+  ComponentFixture
+} from '@angular/core/testing';
+
+import {
+  By
+} from '@angular/platform-browser';
+
+import {
+  SkyAppTestUtility
+} from '@skyux-sdk/testing';
+
+import {
+  SkyDropdownComponent
+} from '../../modules/dropdown/dropdown.component';
+
+import {
+  SkyPopoversFixtureDropdownItem
+} from './popovers-fixture-dropdown-item';
+
+import {
+  SkyPopoversFixtureDropdownMenu
+} from './popovers-fixture-dropdown-menu';
+
+import {
+  SkyPopoversFixtureDropdown
+} from './popovers-fixture-dropdown';
+
+/**
+ * Allows interaction with a SKY UX dropdown component just as a user would.
+ * By using the fixture API, a test insulates itself against updates to the internals
+ * of a component, such as changing its DOM structure.
+ */
+export class SkyDropdownFixture {
+
+  private debugEl: DebugElement;
+
+  /**
+   * Retrieves information about the dropdown component.
+   */
+  public get dropdown(): SkyPopoversFixtureDropdown {{
+    const button = this.buttonDebugElement;
+
+    if (!button) {
+      return;
+    }
+
+    const buttonCmp = button.componentInstance as SkyDropdownComponent;
+
+    return {
+      buttonStyle: buttonCmp.buttonStyle,
+      buttonType: buttonCmp.buttonType,
+      disabled: buttonCmp.disabled,
+      dismissOnBlur: buttonCmp.dismissOnBlur,
+      horizontalAlignment: buttonCmp.horizontalAlignment,
+      label: buttonCmp.label,
+      title: buttonCmp.title
+    };
+  }}
+
+  /**
+   * Returns the dropdown button's text.
+   */
+  public get dropdownButtonText(): string {
+    return this.buttonDebugElement.nativeElement.innerText.trim();
```

Comment:
We have a method in `SkyAppTestUtility` that will get the text of an element and trim it that you can use here.

Reply:
Good tip. Done.